sdk/test/logs/simple_log_processor_test.cc

```diff
@@ -0,0 +1,180 @@
+#include "opentelemetry/sdk/logs/simple_log_processor.h"
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <thread>
+
+using namespace opentelemetry::sdk::logs;
+using opentelemetry::logs::LogRecord;
+
+/*
+ * A test exporter that can return a vector of all the records it has received,
+ * and keep track of the number of times its Shutdown() function was called.
+ * If shutdown succeeded, it returns true.
+ */
+class TestExporter final : public LogExporter
+{
+public:
+  // Initialize a test exporter with 3 pointer parms that can be accessed outside of this class
+  TestExporter(int *shutdown_counter,
+               std::shared_ptr<std::vector<std::string>> logs_received,
+               int *batch_size_received)
+      : shutdown_counter_(shutdown_counter),
+        logs_received_(logs_received),
+        batch_size_received(batch_size_received)
+  {}
+
+  // Stores all the log records this exporter receives to an internal list of log records
+  ExportResult Export(const std::vector<std::unique_ptr<LogRecord>> &records) noexcept override
+  {
+    *batch_size_received = records.size();
+    for (auto &record : records)
+    {
+      // Store the log record name record in the records_received vector
+      if (record != nullptr)
+      {
+        logs_received_->push_back(record->name.data());
+      }
+    }
+    return ExportResult::kSuccess;
+  }
+
+  // Increment the shutdown counter everytime this method is called
+  bool Shutdown(std::chrono::microseconds timeout) noexcept override
+  {
+    *shutdown_counter_ += 1;
+    return true;
+  }
+
+private:
+  int *shutdown_counter_;
+  std::shared_ptr<std::vector<std::string>> logs_received_;
+  int *batch_size_received;
+};
+
+// Tests whether the simple processor successfully creates a batch of size 1
+// and whether the contents of the record is sent to the exporter correctly
+TEST(SimpleLogProcessorTest, SendReceivedLogsToExporter)
+{
+  // Create a new TestExporter
+  std::shared_ptr<std::vector<std::string>> logs_received(new std::vector<std::string>);
+  int batch_size_received = 0;
+
+  std::unique_ptr<TestExporter> exporter(
+      new TestExporter(nullptr, logs_received, &batch_size_received));
+
+  // Create a simple processor with the previous TestExporter attached
+  SimpleLogProcessor processor(std::move(exporter));
+
+  // Send some log records to the processor (which should then send to the TestExporter)
+  const int num_logs = 5;
+  for (int i = 0; i < num_logs; i++)
+  {
+    // Create some sample log records to send to processor
+    auto record = std::unique_ptr<LogRecord>(new LogRecord());
+
+    // Set record name and ensure to_string(i) works properly
+    std::string tmp = "Log name" + std::to_string(i);
+    EXPECT_EQ("Log name" + std::to_string(i), tmp);
```

Comment:
probably don't need this, since checking record->name for same value in a couple lines.